File: fathersDay/Dsa/Day 23 (434. Number of Segments in a String)/Solution.py
class Solution:
    def NumberOfSegments(self, s: str) -> int:
        segments = s.split()
        return len(segments)
solution=Solution()
# Example usage:
s = "Hello, my name is John"
result = solution.NumberOfSegments(s)
print(result)  # Output: 5, as there are five segments in the string

File: fathersDay/Dsa/Day 23 (434. Number of Segments in a String)/Solution.java
class Solution {
    public int NumberOfSegments(String s) {
        String trimmed = s.trim();
        if (trimmed.isEmpty()) {
            return 0;
        }
        return trimmed.split("\\s+").length;
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        String s = "Hello, my name is John";
        int result = solution.NumberOfSegments(s);
        System.out.println(result); // Output: 5
    }
}
